plugins/teststeps/awsfileupload/awsFileUpload.go

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package awsfileupload
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/facebookincubator/contest/pkg/event"
+	"github.com/facebookincubator/contest/pkg/event/testevent"
+	"github.com/facebookincubator/contest/pkg/target"
+	"github.com/facebookincubator/contest/pkg/test"
+	"github.com/facebookincubator/contest/pkg/xcontext"
+	"github.com/facebookincubator/contest/plugins/teststeps"
+)
+
+// Name is the name used to look this plugin up.
+var Name = "awsFileUpload"
+
+// Event names for this plugin.
+const (
+	EventURLtoReport = event.Name("URLtoReport")
+)
+
+// Events defines the events that a TestStep is allow to emit
+var Events = []event.Name{
+	EventURLtoReport,
+}
+
+// FileUpload is used to retrieve all the parameter, the plugin needs.
+type FileUpload struct {
+	path            *test.Param // Path to file that shall be uploaded
+	filename        *test.Param // Filename to file that shall be uploaded
```

Comment:
can this be joined to the `path` variable above? Then you can use `filepath.Base` and `filepath.Dir` to split it into its components in the code below

Reply:
I think it is fine with localpath and filename. The validation is very handy like this.